book/crud.rst

```diff
@@ -0,0 +1,477 @@
+Creating a simple CRUD
+======================
+
+When you deal with your own entities you usually want to create so-called CRUDs for them. CRUD is
+an acronym for **C**\ reate, **R**\ ead, **U**\ pdate and **D**\ elete.
+
+Basically, you will want to create controllers for the following actions:
+
+* :ref:`Display a filterable list of objects. <book-crud-grid>`
+* :ref:`Create forms to create new objects and update existing ones. <book-crud-create-update>`
+* :ref:`View all the details of a single stored object. <book-crud-view>`
+* :ref:`Add a way to delete an entity. <book-crud-delete>`
+
+This chapter assumes that you want to manage the car pool of your company. You want to store the
+car model, the number of seats, the date it was purchased and the date when it needs to be returned
+to the leasing company. The ``Vehicle`` entity class can look like this:
+
+.. code-block:: php
+    :linenos:
+
+    // src/InventoryBundle/Entity/Vehicle.php
+    namespace InventoryBundle\Entity;
+
+    use Doctrine\ORM\Mapping as ORM;
+
+    /**
+     * @ORM\Entity
+     * @ORM\Table(name="vehicle")
+     */
+    class Vehicle
+    {
+        /**
+         * @ORM\Id
+         * @ORM\GeneratedValue(strategy="AUTO")
+         * @ORM\Column(type="integer")
+         */
+        private $id;
+
+        /**
+         * @ORM\Column(type="string", length=100)
+         */
+        private $model;
+
+        /**
+         * @ORM\Column(type="integer")
+         */
+        private $seats;
+
+        /**
+         * @ORM\Column(type="date")
```

Comment:
For camel-cased fields, it's a general practice to specify the name of the column, e.g.:
```php
    /**
     * @ORM\Column(name="bought_at", type="date")
     */
```